Src/Support/Google.Apis.Auth/OAuth2/AwsExternalAccountCredential.cs

```
@@ -0,0 +1,168 @@
+        /// </summary>
+        internal string SecurityCredentialsUrl { get; }
+
+        /// <summary>
+        /// If present, a session token fetched from this URL should be used when making
+        /// requests to the metadata server.
+        /// </summary>
+        internal string ImdsV2SessionTokenUrl { get; }
+
+        internal AwsExternalAccountCredential(Initializer initializer) : base(initializer)
+        {
+            RegionUrl = initializer.RegionUrl;
+            RegionalCredentialVerificationUrl = initializer.RegionalCredentialVerificationUrl;
+            SecurityCredentialsUrl = initializer.SecurityCredentialsUrl;
+            ImdsV2SessionTokenUrl = initializer.ImdsV2SessionTokenUrl;
+        }
+
+        /// <inheritdoc/>
+        private protected override GoogleCredential WithoutImpersonationConfigurationImpl() =>
+            ServiceAccountImpersonationUrl is null
+            ? new GoogleCredential(this)
+            : new GoogleCredential(new AwsExternalAccountCredential(new Initializer(this)
+            {
+                ServiceAccountImpersonationUrl = null
+            }));
+
+        /// <inheritdoc/>
+        protected async override Task<string> GetSubjectTokenAsyncImpl(CancellationToken taskCancellationToken)
+        {
+            // We might not need to use the metadata server at all,
+            // but initializing this client here triggers no requests, etc.
+            // and makes the code cleaner.
+            var metadataClient = new AwsMetadataServerClient(ImdsV2SessionTokenUrl, HttpClient);
+
+            var awsRegion = await AwsRegion.FetchAsync(metadataClient, RegionUrl).ConfigureAwait(false);
+            var awsSecurityCredentials = await AwsSecurityCredentials.FetchAsync(metadataClient, SecurityCredentialsUrl).ConfigureAwait(false);
+            var regionalizedVerificationUrl = RegionalCredentialVerificationUrl.Replace(RegionalCredentialVerificationUrlRegionPlaceholder, awsRegion.Region);
```

I think this could throw a NullReferenceException - we're never checking it's non-null, unless I've missed something. It would be better to check it and throw an appropriate exception than let the NRE propagate.